aiida/orm/implementation/sqlalchemy/logs.py

```diff
@@ -113,9 +113,11 @@ def delete_many(self, filters):
         """
         Delete all log entries in the table
         """
+        session = get_scoped_session()
+
         if not filters:
             for entry in models.DbLog.query.all():
                 entry.delete()
-            get_scoped_session().commit()
+            session.commit()
```

Comment:
I would put his in a transaction
https://docs.sqlalchemy.org/en/latest/orm/session_transaction.html

```
Session = sessionmaker(bind=engine, autocommit=True)
session = Session()
session.begin()
try:
    item1 = session.query(Item).get(1)
    item2 = session.query(Item).get(2)
    item1.foo = 'bar'
    item2.bar = 'foo'
    session.commit()
except:
    session.rollback()
    raise
```

Reply:
I will do the following:

```python
session = get_scoped_session()

try:
    # do the if, else statement
except Exception as exc:
    session.rollback()
    raise exceptions.DbContentError(exc)
```

Reply:
Something that I missed when I did my previous comment is that a commit in this method (at least in SQLA), will commit everything until this point. This is something that we don't want during import because it will commit data in the middle of the import.

A solution is nested transactions. Martin has implemented a context manager for transactions that seems to correctly handle nested transactions for both backends.
aiida.orm.implementation.backends.Backend#transaction

This could be used which will also allow you to have less code overlap in these two delete_many method.

aiida.orm.implementation.backends.Backend#transaction could also be used (please test) to replace the current transactions in import_data_dj/import_data_sqla